sympy/matrices/matrices.py

```diff
@@ -1594,6 +1596,12 @@ def __pow__(self, num):
                     return self._matrix_pow_by_recursion(num)
             return self._matrix_pow_by_recursion(num)
         elif isinstance(num, (Expr, float)):
+            num = sympify(num)
+            if (num >= 0) == True and not num % 1 == 0:
+                z = self.jordan_form()[1]
```

Comment:
>If that is done, it is not necessary to ask if ```num``` is an ```int``` or a ```float```

In case of ```float``` will it help? But yes ```isinstance(num, (int, Integer))``` can be replaced.
```
In [13]: not S(a).is_integer
Out[13]: True

In [14]: not S(50.2).is_integer
Out[14]: True

In [15]: S(a).is_integer
```